dm/pkg/binlog/pos_finder_test.go

```diff
@@ -0,0 +1,447 @@
+	buf.Write(xidEvent.RawData)
+
+	// second transaction, timestamp of all events = beforeTime.Add(3 * time.Second)
+	generator.LatestPos = xidEvent.Header.LogPos
+	dmlEvents, data, _ := generator.GenDMLEvents(replication.WRITE_ROWS_EVENTv2, insertDMLData, ts)
+	buf.Write(data)
+
+	c.Assert(dmlEvents[len(dmlEvents)-1].Header.LogPos, Equals, uint32(buf.Len()))
+	_ = os.WriteFile(path.Join(relayDir, "mysql-bin.000001"), buf.Bytes(), 0644)
+
+	{
+		tcctx := tcontext.NewContext(context.Background(), log.L())
+		finder := NewLocalBinlogPosFinder(tcctx, false, flavor, relayDir)
+		location, posType, err := finder.FindByTs(ts)
+		c.Assert(err, IsNil)
+		// start of second transaction
+		c.Assert(location.Position, Equals, mysql.Position{"mysql-bin.000001", xidEvent.Header.LogPos})
+		c.Assert(location.GTIDSetStr(), Equals, "")
+		c.Assert(posType, Equals, InRangeBinlogPos)
+	}
+}
+
+func (t *testPosFinderSuite) TestMySQL56NoGTID(c *C) {
+	flavor := "mysql"
+	relayDir := c.MkDir()
+	beforeTime := time.Now()
+	latestGTIDStr := "ffffffff-ffff-ffff-ffff-ffffffffffff:1"
+
+	generator, _ := event.NewGeneratorV2(flavor, "5.6.0", latestGTIDStr, false)
+
+	file1Events, data := genBinlogFile(generator, beforeTime, "mysql-bin.000002")
+	c.Assert(len(file1Events), Equals, 11)
+	_ = os.WriteFile(path.Join(relayDir, "mysql-bin.000001"), data, 0644)
+	file2Events, data := genBinlogFile(generator, beforeTime.Add(5*time.Second), "mysql-bin.000003")
+	c.Assert(len(file2Events), Equals, 11)
+	_ = os.WriteFile(path.Join(relayDir, "mysql-bin.000002"), data, 0644)
+	file3Events, data := genBinlogFile(generator, beforeTime.Add(10*time.Second), "mysql-bin.000004")
+	c.Assert(len(file3Events), Equals, 11)
+	_ = os.WriteFile(path.Join(relayDir, "mysql-bin.000003"), data, 0644)
+
+	tcctx := tcontext.NewContext(context.Background(), log.L())
+	{
+		var targetEventStart uint32
+		var targetEvent *replication.BinlogEvent
+		for _, ev := range file1Events {
+			if e, ok := ev.Event.(*replication.QueryEvent); ok && string(e.Query) == "BEGIN" {
+				targetEvent = ev
+				break
+			}
+			targetEventStart = ev.Header.LogPos
+		}
+		finder := NewLocalBinlogPosFinder(tcctx, false, flavor, relayDir)
+		location, posType, err := finder.FindByTs(int64(targetEvent.Header.Timestamp))
+		c.Assert(err, IsNil)
+		c.Assert(location.Position, Equals, mysql.Position{"mysql-bin.000001", targetEventStart})
+		c.Assert(location.GTIDSetStr(), Equals, "")
+		c.Assert(posType, Equals, InRangeBinlogPos)
+	}
+	{
+		var targetEventStart = file2Events[len(file2Events)-1].Header.LogPos
+		finder := NewLocalBinlogPosFinder(tcctx, false, flavor, relayDir)
+		location, posType, err := finder.FindByTs(int64(file3Events[0].Header.Timestamp))
+		c.Assert(err, IsNil)
+		c.Assert(location.Position, Equals, mysql.Position{"mysql-bin.000002", targetEventStart})
+		c.Assert(location.GTIDSetStr(), Equals, "")
+		c.Assert(posType, Equals, InRangeBinlogPos)
+	}
+	{
+		var targetEventStart uint32
```

Comment:
can you please add some comments to explian the difference betweed this three `fileEvents` and `targetEvent` and what is the different purepose of this tree test?

Reply:
try read the part where `fileEvents` is generated, i think it's quite verbose from there.

i think the variable name is quite self-explanatory, don't follow what's misleading you, or what you suggest the variable name should be.